website/src/components/lesson.css.ts

```diff
@@ -0,0 +1,117 @@
+import { globalStyle, style } from "@vanilla-extract/css"
+import { important, padding } from "polished"
+import { Styles } from "polished/lib/types/style"
+import {
+  colors,
+  hsize,
+  hspace,
+  space,
+  thickness,
+  vspace,
+} from "src/style/constants"
+import { marginY, paddingX } from "src/style/utils"
+
+// Parent class defining a Wordpress page.
+// All classes defined in this file begin with this generated className to control the scope of these styles.
+export const lesson = style({})
+
+const mobileQuery = "screen and (max-width: 52em)"
+
+//// INPUT STYLES
+const mobileInputSize = `calc(1rem * 2)`
+
+// On mobile, increases width and height of input boxes for better mobile visibility.
+globalStyle(`${lesson} input`, {
+  "@media": {
+    [mobileQuery]: important({
+      width: hsize.full,
+      height: mobileInputSize,
+    }),
+  },
+})
+
+// A global style that styles fill in the blank sentences where a word
+// appears below the input box.
+globalStyle(`${lesson} .fill-blank span`, {
+  display: "inline-flex",
+  flexDirection: "column",
+  alignItems: "center",
+  ...paddingX(hspace.halfEdge),
+})
+
+//// IMAGE AND CAPTION STYLES
+const figure = `${lesson} figure`
+// Gives flexbox items (image and caption) a column look only on mobile.
+globalStyle(figure, {
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "center",
+  textAlign: "left",
+  gap: `calc(${space.large} * 2)`,
+
+  "@media": {
+    [mobileQuery]: important({
+      flexDirection: "column",
+      gap: 0,
+    }),
+  },
+})
+
+// Styles the caption text of figures.
+globalStyle(`${figure} figcaption > *`, {
+  color: colors.primary,
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "start",
+})
+
+/// TABLE STYLES
+// Styles rows of a table.
+// On mobile, since row width is limited, convert each row into a "column-like" element
+// and space out their margins.
+globalStyle(`${lesson} table tr`, {
+  "@media": {
+    [mobileQuery]: important({
+      display: "flex",
+      flexDirection: "column",
+      border: `${thickness.thin} solid ${colors.text}`,
+      ...(marginY(vspace.large) as Styles),
```

Comment:
I'm surprised this `as` clause is necessary, it wasn't passing typecheck? Later on, `padding` seems to work fine so maybe revise the type of the `marginY` function.

Reply:
Do you mean revise the type that `marginY` returns? Since it looks like the `padding` functions that work return a `Styles` instead of `StyleRules` which allows them to work

Reply:
Let's try this out chanmi, and deploy changes after @GracefulLemming reviews.